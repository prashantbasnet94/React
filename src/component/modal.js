import React, { useState } from 'react';
import { Button, Modal } from 'antd';

export const ModalCmp = () => {
  const [isModalOpen, setIsModalOpen] = useState(false);

  const showModal = () => {
    setIsModalOpen(true);
  };

  const handleOk = () => {
    setIsModalOpen(false);
  };

  const handleCancel = () => {
    setIsModalOpen(false);
  };

  return (
    <>
      <Button type="primary" onClick={showModal}  data-testid="openModal"  >Open Modal</Button>
      <Modal title="Basic Modal" open={isModalOpen} onOk={handleOk} onCancel={handleCancel} role ='popup'>
        <p>Some contents...</p>
        <p>Some contents...</p>
        <p>Some contents...</p>
      </Modal>
    </>
  );
};

 