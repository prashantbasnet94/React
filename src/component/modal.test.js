import { ModalCmp } from "./modal"
import renderer from 'react-test-renderer';

describe("testing modal pop up", () => {
    let
        component = renderer.create(<ModalCmp />),
        mycmp = component.toJSON()

    expect(mycmp).toMatchSnapshot();

    it('Should have a button and not disabled', () => {
        expect(mycmp.type).toEqual('button')
        expect(mycmp?.props?.disabled).toBeFalsy()
    })
})