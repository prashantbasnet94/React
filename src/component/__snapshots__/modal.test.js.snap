// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<button
  className="ant-btn css-dev-only-do-not-override-1n7nwfa ant-btn-primary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <span>
    Open Modal
  </span>
</button>
`;
